rails/client/app/bundles/kaiju/components/Component/Component.jsx

```diff
@@ -1,31 +1,59 @@
 import React from 'react';
 import Base from 'terra-base';
 import PropTypes from 'prop-types';
+import ajax from 'superagent';
+import classNames from 'classnames/bind';
 import { Provider } from 'react-redux';
 import configureStore from './store/componentStore';
 import ComponentContainer from './containers/ComponentContainer';
+import Spinner from '../common/Spinner/Spinner';
 import distpatcher from './utilities/dispatcher';
 import initializeDrag from './utilities/drag';
+import styles from './Component.scss';
+
+const cx = classNames.bind(styles);
 
 const propTypes = {
   /**
-   * The Component Identifer
+   * The Component identifer.
    */
   id: PropTypes.string.isRequired,
+  /**
+   * The Component url.
+   */
+  url: PropTypes.string.isRequired,
 };
 
 class Component extends React.Component {
   constructor(props) {
     super(props);
-    this.store = configureStore(this.props);
+
+    this.state = { loading: true };
   }
 
   componentDidMount() {
-    distpatcher(this.store, this.props.id);
     initializeDrag();
+
+    this.request = ajax
+      .get(this.props.url)
+      .set('Accept', 'application/json')
+      .end((error, response) => {
+        if (response) {
```

Comment:
You should handle the error

Reply:
I added this check only for the case where the request is aborted if the iframe is unmounted during the request, such as when a user is clicking between tabs without waiting until the tab has finished loading. I can split the logic to handle this case explicitly, but it would be an empty block. I was thinking when we create a wrapper for axio's / super agent we would create a more robust error handling system for all requests.